lighthouse-core/lib/url-shim.js

```diff
@@ -87,6 +87,26 @@ class URLShim extends URL {
     }
   }
 
+  /**
+   * Check if rootDomains matches
+   *
+   * @param {string} urlA
+   * @param {string} urlB
+   */
+  static rootDomainsMatch(urlA, urlB) {
+    const urlAInfo = new URL(urlA);
+    const urlBInfo = new URL(urlB);
+
+    if (!urlAInfo.hostname || !urlBInfo.hostname) {
+      return false;
+    }
+
+    const urlARootDomain = urlAInfo.hostname.split('.').slice(-2).join('.');
```

Comment:
this is called "tld plus one". let's use that term for the method name and here.

let's add a few whitelisted exceptions like '.co.uk'. (country-code second level domains).